File: day06-数组-元素移出和数组的平方/977.有序数组的平方/977.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortedSquares = function (nums) {
  let right = nums.length - 1;
  let left = 0;
  let pos = right;
  let ans = new Array(pos + 1);
  while (left <= right) {
    if (nums[left] * nums[left] > nums[right] * nums[right]) {
      ans[pos] = nums[left] * nums[left];
      left++;
    } else {
      ans[pos] = nums[right] * nums[right];
      right--;
    }
    pos--;
  }
  return ans;
};

File: day06-数组-元素移出和数组的平方/977.有序数组的平方/977.cpp
#include <iostream>
#include <vector>
using namespace std;
class Solution
{
public:
    vector<int> sortedSquares(vector<int> &nums)
    {
        int left = 0;
        int right = nums.size() - 1;
        int pos = right;
        vector<int> ans(nums.size());
        while (left <= right)
        {
            if (nums[left] * nums[left] >= nums[right] * nums[right])
            {
                nums[pos] = nums[left] * nums[left];
                left++;
            }
            else
            {
                nums[pos] = nums[right] * nums[right];
                right--;
            }
            pos--;
        }
        return nums;
    }
};